platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_configurations.adoc

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-configurations]]
+= 2. Dependency Configurations
+
+Every dependency declared for a Gradle project applies to a specific scope.
+
+For example, some dependencies should be used for compiling source code whereas others only need to be available at runtime:
+
+[source, kotlin]
+.build.gradle.kts
+----
+dependencies {
+    implementation("com.google.guava:guava:30.0-jre") // Needed to compile and run the app
+    runtimeOnly("org.slf4j:slf4j-simple:2.0.13") // Only needed at runtime
+}
+----
+
+Dependency configurations are a way to define different sets of dependencies for different purposes within a project.
+They determine how and when dependencies are used in various stages of the build process.
+
+Configurations are a fundamental part of dependency resolution in Gradle.
+
+[[sec:what-are-dependency-configurations]]
+== Understanding dependency configurations
+
+Gradle represents the scope of a dependency with the help of a link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[Configuration].
+Every configuration can be identified by a unique name.
+
+Many Gradle plugins add pre-defined configurations to your project.
+
+The Java Library plugin is used to define a project that produces a Java library.
+The plugin adds many configurations.
+These configurations represent the various classpaths needed for source code compilation, executing tests, and more:
+
+[cols="~,~,~"]
+|===
+|Configuration Name |Description |Used to:
+
+|`implementation`
```

Comment:
Do we really want to duplicate all this information that is already available here, especially outside of a JVM chapter: https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph

Reply:
If we _are_ going to list these here, I think we should include `api` and `compileOnlyApi`, which are included with the `java-library` plugin -- the most popular JVM plugin

Reply:
Removed the configs.